tests/lib/rules/require-computed-macros.js

```diff
@@ -287,5 +287,56 @@ ruleTester.run('require-computed-macros', rule, {
       output: "class Test { @computed.mapBy('children', 'age') someProp }",
       errors: [{ message: ERROR_MESSAGE_MAP_BY, type: 'MethodDefinition' }],
     },
+
+    // Self Referential
```

Comment:
I think we're adding a few redundant test cases as well as missing some test cases.

We need the following test cases basic test cases which test a simple single-arg return expression that will be converted into the `reads` macro:
* `class Test { @computed() get foo() { return this.foo; } }` 
* `class Test { foo = computed(function() { return this.foo; }) }` // non-decorator style
* `import EmberObject from '@ember/object'; EmberObject.extend({ foo: computed(function() { return this.foo; }) }` // classic class

And we need to test other types of expressions that will be converted into other macros:
* `class Test { @computed() get a() { return this.a && this.b; } }` // logical 
*`class Test { @computed() get a() { return this.a > this.b; } }` // binary 
*`import EmberObject from '@ember/object'; EmberObject.extend({ computed(function() { return this.chores.filterBy('done', true); })` // function 

We don't need to test anything related to the dependent key arguments (`computed('this-dependent-key-does-not-matter', function () {})`) since those end up ignored anyway.

Reply:
The redundancy is intentional, it covers both forms of classic and both forms of class based declarations both with and without args being supplied to the computed as dependent keys (since I encountered errors from conversion from both in the wild).

Can add tests for a few of the others.

Reply:
A comment next to each test case explaining what variation it is testing would help.